parser/src/python.lalrpop

```diff
@@ -787,6 +787,14 @@ MulOp: ast::Operator = {
     "@" => ast::Operator::MatMult,
 };
 
+NamedExpression: ast::Expression = {
+    <e1:Expression> <location:@L> ":=" <e2:Expression> => ast::Expression {
+        location,
+        node: ast::ExpressionType::NamedExpression { target: Box::new(e1), value: Box::new(e2) }
+    },
+    NamedExpression,
+};
```

Comment:
Probably this is wrong.  I'm getting ```LalrpopError::UnrecognizedToken```->```ParseErrorType::UnrecognizedToken```

```
>>>>> if any(len(longline := line) >= 100 for line in lines):
SyntaxError: 
if any(len(longline := line) >= 100 for line in lines):
                    ↑
                    Got unexpected token ':='
```

Reply:
It looks like you might want to use `TestOrStarExpr`, as that's what the assignment statement does: https://github.com/RustPython/RustPython/blob/ed37c61df7eed1ada6e1a063720a20a0eb6ec58f/parser/src/python.lalrpop#L88

Reply:
It didn't work unfortunately.  Might still be doing something wrong, idk.  I'll take another shot after New Years and do some deeper digging.

Reply:
Please check with the grammar on this page: https://docs.python.org/3/reference/grammar.html?highlight=grammar that contains useful tips on what to modify.